Firestore/core/src/local/memory_remote_document_cache.cc

```diff
@@ -55,31 +54,32 @@ void MemoryRemoteDocumentCache::Remove(const DocumentKey& key) {
   docs_ = docs_.erase(key);
 }
 
-absl::optional<MaybeDocument> MemoryRemoteDocumentCache::Get(
-    const DocumentKey& key) {
+MutableDocument MemoryRemoteDocumentCache::Get(const DocumentKey& key) {
   const auto& entry = docs_.get(key);
-  return entry ? entry->first : absl::optional<MaybeDocument>();
+  // Note: We create an explicit copy to prevent modifications or the backing
+  // data.
+  return entry ? entry->first : MutableDocument::InvalidDocument(key);
 }
 
-OptionalMaybeDocumentMap MemoryRemoteDocumentCache::GetAll(
+MutableDocumentMap MemoryRemoteDocumentCache::GetAll(
     const DocumentKeySet& keys) {
-  OptionalMaybeDocumentMap results;
+  MutableDocumentMap results;
   for (const DocumentKey& key : keys) {
     // Make sure each key has a corresponding entry, which is nullopt in case
     // the document is not found.
     // TODO(http://b/32275378): Don't conflate missing / deleted.
-    results = results.insert(key, Get(key));
+    results = results.insert(key, std::move(Get(key)));
```

Comment:
You don't need to move a temporary -- it is an rvalue and will trigger the move constructor anyway.